hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTenantConfigurationManagement.java

```diff
@@ -133,45 +143,65 @@ public <T> T getGlobalConfigurationValue(final String configurationKeyName, fina
     }
 
     @Override
-    @CacheEvict(value = "tenantConfiguration", key = "#configurationKeyName")
-    @Transactional
-    @Retryable(include = {
-            ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public <T extends Serializable> TenantConfigurationValue<T> addOrUpdateConfiguration(
```

Comment:
Without annotation of this method this wont work as expected. The spring proxy mechanism won't apply annotations of internally called method. See [Understanding the correct use of Declarative Transaction](https://www.logicbig.com/tutorials/spring-framework/spring-data-access-with-jdbc/correct-use-of-declarative-transaction.html) 
You should have _@Transactional_ and _@Retriable_ on both methods.
Maybe the clear approach would be to have third private method that will do the work and both methods to be annotated and directly to call the private method.